pilot/pkg/model/push_context.go

```diff
@@ -85,6 +90,13 @@ type PushContext struct {
 	initDone bool
 }
 
+type processedDestRules struct {
+	// List of dest rule hosts. We match with the most specific host first
+	hosts []Hostname
```

Comment:
TODO: Is it sorted ? Should we use a map[] for exact matches, and list only for prefixes ?  ( no need to change in this PR, but would be good to know the answer and add a TODO if it makes sense )

Reply:
There is no data structure change in this PR. i.e. we have the same data structures we had earlier. Sorted array of hosts, map of hostname->destRule. All the optimizations were done a while ago